core/src/main/scala/org/apache/spark/TaskEndReason.scala

```diff
@@ -217,11 +217,14 @@ case class TaskCommitDenied(
  * the task crashed the JVM.
  */
 @DeveloperApi
-case class ExecutorLostFailure(execId: String, isNormalExit: Boolean = false)
-  extends TaskFailedReason {
+case class ExecutorLostFailure(execId: String, isNormalExit: Boolean = false, reason:
+Option[String]) extends TaskFailedReason {
```

Comment:
Indentation here is weird. I'd just break the parameters following the usual pattern:

    case class ExecutorLostFailure(
        execId: String,
        isNormalExit: Boolean = false,
        reason: Option[String]) extends TaskFailedReason {